tests/core/process.c

```diff
@@ -710,11 +718,43 @@ store_export_address_if_mach_msg (const GumExportDetails * details,
 
 #endif
 
+static GThread *
+new_sleeping_dummy_thread_sync (volatile gboolean * done)
+{
+  TestThreadSyncData sync_data;
+  GThread * thread;
+
+  g_cond_init (&sync_data.cond);
+  g_mutex_init (&sync_data.mutex);
+  g_mutex_lock (&sync_data.mutex);
+  sync_data.started = FALSE;
+  sync_data.done = done;
+
+  thread = g_thread_new ("process-test-sleeping-dummy", sleeping_dummy,
+      &sync_data);
+  
+  while (! sync_data.started)
+    g_cond_wait (&sync_data.cond, &sync_data.mutex);
+
+  g_mutex_unlock (&sync_data.mutex);
+
+  g_cond_clear (&sync_data.cond);
+  g_mutex_clear (&sync_data.mutex);
+
+  return thread;
+}
+
 static gpointer
 sleeping_dummy (gpointer data)
 {
-  volatile gboolean * done = (gboolean *) data;
+  TestThreadSyncData * sync_data = (TestThreadSyncData *)data;
```

Comment:
Still missing space after cast. But let’s drop the cast and not copy the mistakes of my younger self. ;)